src/shell.cpp

```diff
@@ -0,0 +1,58 @@
+#include "shell.h"
+
+void Shell::runCommand(const std::string& command, bool inBackground) {
+    std::vector<std::string> args = parser.split(command, ' ');
+    std::vector<char*> cargs;
+    for (const auto& arg : args) {
+        cargs.push_back(const_cast<char*>(arg.c_str()));
```

Comment:
This is an interesting point. 
Can you explain why you need to use `const_cast` and not `std::vector<const char*>`?